File: example/src/main/java/com/example/socialview/MainActivity.java
package com.example.socialview;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import butterknife.BindViews;

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
public class MainActivity extends BaseActivity implements View.OnClickListener {

    @BindViews({
            R.id.button_main_example1,
            R.id.button_main_example2,
            R.id.button_main_example3,
            R.id.button_main_example4
    }) Button[] buttons;

    @Override
    protected int getContentView() {
        return R.layout.activity_main;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        for (Button button : buttons)
            button.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.button_main_example1:
                startActivity(new Intent(this, Example1Activity.class));
                break;
            case R.id.button_main_example2:
                startActivity(new Intent(this, Example2Activity.class));
                break;
            case R.id.button_main_example3:
                startActivity(new Intent(this, Example3Activity.class));
                break;
            case R.id.button_main_example4:
                startActivity(new Intent(this, Example4Activity.class));
                break;
        }
    }
}

File: example/src/main/java/com/example/socialview/MainActivity.kt
package com.example.socialview

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

/**
 * @author Hendra Anggrian (com.hendraanggrian@gmail.com)
 */
class MainActivity : AppCompatActivity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        for (button in arrayOf(button1, button2, button3, button4)) {
            button.setOnClickListener(this)
        }
    }

    override fun onClick(v: View) {
        startActivity(Intent(this, when (v.id) {
            R.id.button1 -> Example1Activity::class.java
            R.id.button2 -> Example2Activity::class.java
            R.id.button3 -> Example3Activity::class.java
            else -> Example4Activity::class.java
        }))
    }
}